src/main/java/com/sap/cloudfoundry/client/facade/adapters/RawCloudPackage.java

```diff
@@ -31,26 +34,34 @@ private static Status parseStatus(Package resource) {
         return parseEnum(resource.getState(), Status.class);
     }
 
-    private static CloudPackage.Data parseData(Package resource) {
+    private static CloudPackage.PackageData parseData(Package resource) {
         if (resource.getType() == PackageType.BITS) {
             return parseBitsData((BitsData) resource.getData());
         }
-        return null;
+        return parseDockerData((DockerData) resource.getData());
     }
 
-    private static CloudPackage.Data parseBitsData(BitsData data) {
-        return ImmutableCloudPackage.ImmutableData.builder()
-                                                  .checksum(parseChecksum(data.getChecksum()))
-                                                  .error(data.getError())
+    private static CloudPackage.PackageData parseBitsData(BitsData data) {
+        return ImmutableBitsData.builder()
+                                .checksum(parseChecksum(data.getChecksum()))
+                                .error(data.getError())
+                                .build();
+    }
+
+    private static com.sap.cloudfoundry.client.facade.domain.BitsData.Checksum parseChecksum(Checksum checksum) {
```

Comment:
Can you import this

Reply:
No because the v3 Checksum object is already imported :(